### PR TITLE
Balance druid implementation

### PR DESCRIPTION
**Pull request by stn-ms**
some bug fixes to stuff that was reported to me directly.

Also split Moonfire into a dot spell and impact spell and added Sunfire. If one is already present, the other one should overwrite it.
Moved a few of the glyphs to a new glyphs.go
Added mastery bonuses to eclipses.
Insect Swarm ticks not critting.
A more advanced APL (thanks Shantie <3)

### CODE REVIEW

File: sim/druid/eclipse.go [hunk 1]
```diff
@@ -56,15 +56,29 @@ func (druid *Druid) EnableEclipseBar() {
 }
 
 func (druid *Druid) RegisterEclipseAuras() {
+	masteryBonus := 0.16 + druid.GetMasteryPoints()*0.02
```
Comment (rosenrusinov) [inline sim/druid/eclipse.go:59]: You dont have mastery fully initialized during registration of spells/auras. You need to update the value again in the auras OnGain methods before activating the mods. Also you should register to the On Mastery Changed callback to update the spell mods values when mastery changes due to procs

File: sim/druid/moonfire.go [hunk 2]
```diff
@@ -43,29 +31,63 @@ func (druid *Druid) registerMoonfireSpell() {
...
 			},
 			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
 				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeSnapshotCrit)
 			},
 		},
 
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			result := spell.CalcOutcome(sim, target, spell.OutcomeAlwaysHit)
+
+			if result.Landed() {
```
Comment (rosenrusinov) [inline sim/druid/moonfire.go:48]: no point of this if when the outcome is always hit

File: ui/core/proto_utils/action_id.ts [hunk 3]
```diff
@@ -378,6 +378,16 @@ export class ActionId {
 				break;
 			case 'Fulmination':
 				name += ` (${this.tag + 3})`
+			case 'Moonfire':
+				if (this.tag == 1) {
+					name += ' (DoT)'
+				}
+				break;
```
Comment (rosenrusinov) [inline ui/core/proto_utils/action_id.ts:385]: you can have the switch case falltrough to the next case so just putting:
```
case 'Moonfire':
case 'Sunfire':
  dot...
```

**Review by rosenrusinov — COMMENTED** [review]
Only real issue is the mastery related spell mods. Other things are nits

File: sim/druid/eclipse.go (hunk 1 above)
Comment (stn-ms) [reply sim/druid/eclipse.go:59]: I can move it there, but I don't think I need to update it as Eclipse is supposedly snapshotting mastery.

File: ui/core/proto_utils/action_id.ts (hunk 3 above)
Comment (stn-ms) [reply ui/core/proto_utils/action_id.ts:385]: oh true, thanks

File: sim/druid/eclipse.go (hunk 1 above)
Comment (rosenrusinov) [reply sim/druid/eclipse.go:59]: the spell mod should still be properly updated on mastery changes. The dot will still properly snapshot the damage on application but you still have to have the correct value to snapshot from.

Comment (stn-ms) [reply sim/druid/eclipse.go:59]: yep, that's done.

**Pull request merged**